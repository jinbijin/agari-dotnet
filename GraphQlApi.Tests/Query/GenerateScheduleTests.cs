using System.Threading.Tasks;
using GraphQlApi.Tests.Fixture;
using GraphQlApi.Tests.Variables;
using Snapshooter.Xunit;
using Xunit;

namespace GraphQlApi.Tests.Query
{
    public class GenerateScheduleTests : GraphQlApiTest<GenerateScheduleVariables>, IClassFixture<WebApiFixture>
    {
        private const string Query = "query GenerateSchedule($roundCount: Int!, $participantCount: Int!) { generateSchedule(roundCount: $roundCount, participantCount: $participantCount) { rounds { games { participantNrs } } } }";

        public GenerateScheduleTests(WebApiFixture webApiFixture) : base(webApiFixture, Query)
        {
        }

        [Fact(DisplayName = "generateSchedule OK")]
        public Task OkTest() =>
            MatchResponseForRequest(new GenerateScheduleVariables
            {
                roundCount = 4,
                participantCount = 20
            }, Snapshot.FullName());

        [Fact(DisplayName = "generateSchedule Error: not enough participants")]
        public Task NotEnoughParticipantsTest() =>
            MatchResponseForRequest(new GenerateScheduleVariables
            {
                roundCount = 4,
                participantCount = 16
            }, Snapshot.FullName());

        [Fact(DisplayName = "generateSchedule Error: not enough rounds")]
        public Task NotEnoughRoundsTest() =>
            MatchResponseForRequest(new GenerateScheduleVariables
            {
                roundCount = 3,
                participantCount = 20
            }, Snapshot.FullName());

        [Fact(DisplayName = "generateSchedule Error: invalid participant count")]
        public Task InvalidParticipantCountTest() =>
            MatchResponseForRequest(new GenerateScheduleVariables
            {
                roundCount = 4,
                participantCount = 21
            }, Snapshot.FullName());
    }
}
