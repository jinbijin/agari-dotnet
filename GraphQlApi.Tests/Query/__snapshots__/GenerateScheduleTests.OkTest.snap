{
  "ResponseHeaders": {
    "Version": "1.1",
    "Content": {
      "Headers": [
        {
          "Key": "Content-Type",
          "Value": [
            "application/json; charset=utf-8"
          ]
        },
        {
          "Key": "Content-Length",
          "Value": [
            "710"
          ]
        }
      ]
    },
    "StatusCode": "OK",
    "ReasonPhrase": "OK",
    "Headers": [],
    "TrailingHeaders": [],
    "RequestMessage": {
      "Version": "1.1",
      "VersionPolicy": "RequestVersionOrLower",
      "Content": {
        "Headers": [
          {
            "Key": "Content-Type",
            "Value": [
              "application/json; charset=utf-8"
            ]
          },
          {
            "Key": "Content-Length",
            "Value": [
              "252"
            ]
          }
        ]
      },
      "Method": {
        "Method": "POST"
      },
      "RequestUri": "http://localhost/graphql",
      "Headers": [],
      "Properties": {},
      "Options": {}
    },
    "IsSuccessStatusCode": true
  },
  "ResponseContent": {
    "data": {
      "generateSchedule": {
        "rounds": [
          {
            "games": [
              {
                "participantNrs": [
                  0,
                  1,
                  2,
                  3
                ]
              },
              {
                "participantNrs": [
                  4,
                  5,
                  6,
                  7
                ]
              },
              {
                "participantNrs": [
                  8,
                  9,
                  10,
                  11
                ]
              },
              {
                "participantNrs": [
                  12,
                  13,
                  14,
                  15
                ]
              },
              {
                "participantNrs": [
                  16,
                  17,
                  18,
                  19
                ]
              }
            ]
          },
          {
            "games": [
              {
                "participantNrs": [
                  2,
                  4,
                  9,
                  19
                ]
              },
              {
                "participantNrs": [
                  3,
                  6,
                  8,
                  13
                ]
              },
              {
                "participantNrs": [
                  7,
                  10,
                  12,
                  17
                ]
              },
              {
                "participantNrs": [
                  1,
                  11,
                  14,
                  16
                ]
              },
              {
                "participantNrs": [
                  0,
                  5,
                  15,
                  18
                ]
              }
            ]
          },
          {
            "games": [
              {
                "participantNrs": [
                  2,
                  8,
                  15,
                  17
                ]
              },
              {
                "participantNrs": [
                  1,
                  6,
                  12,
                  19
                ]
              },
              {
                "participantNrs": [
                  3,
                  5,
                  10,
                  16
                ]
              },
              {
                "participantNrs": [
                  0,
                  7,
                  9,
                  14
                ]
              },
              {
                "participantNrs": [
                  4,
                  11,
                  13,
                  18
                ]
              }
            ]
          },
          {
            "games": [
              {
                "participantNrs": [
                  2,
                  5,
                  11,
                  12
                ]
              },
              {
                "participantNrs": [
                  6,
                  9,
                  15,
                  16
                ]
              },
              {
                "participantNrs": [
                  0,
                  10,
                  13,
                  19
                ]
              },
              {
                "participantNrs": [
                  3,
                  4,
                  14,
                  17
                ]
              },
              {
                "participantNrs": [
                  1,
                  7,
                  8,
                  18
                ]
              }
            ]
          }
        ]
      }
    }
  }
}
